### ISSUE TITLE
trt_testimg.py如何用

### ISSUE THREAD

**Issue by fanghaiquan1** [PolarisKyle/trt_GridDehazeNet#1]
请问trt_img.py与trt_testimg.py的作用，两者区别，这个trt_img.py程序我可以正常跑通得到好的结果，但是比较耗时。trt_testimg.py得到的结果不是很好，得到的结果图是朦的，不知道什么原因？可能是我没有用对.

from volksdep.converters import load
import pdb
import time
import argparse
from torchvision import transforms
import numpy as np
from PIL import Image
import torch
from torchvision.utils import save_image
import tensorrt as trt
import cv2

trt.init_libnvinfer_plugins(None, "")
trt_model = load('GFPGANv1.4.trt')

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description = "TensorRT do inference")
    parser.add_argument("--batch_size", type=int, default=1, help='batch_size')
    parser.add_argument("--img_path", type=str, default='img1.png', help='cache_file')
    parser.add_argument("--engine_file_path", type=str, default='GFPGANv1.4.trt', help='engine_file_path')
    args = parser.parse_args()

    img = Image.open(args.img_path)
    input_shape = (1, 3, 512, 512)
    transform = transforms.Compose([
        transforms.Resize([input_shape[2], input_shape[3]]),  # [h,w]
        transforms.ToTensor()
        ])
    img = transform(img).unsqueeze(0).cuda()
    # pdb.set_trace()
    # img = img.numpy()
    with torch.no_grad():
        t1 = time.time()
        trt_output = trt_model(img)
        print(time.time()-t1)
        # trt_output = trt_output.squeeze(0)
        # trt_output = trt_output.permute(1, 2, 0)
        # pdb.set_trace()
        # _output = trt_output.detach().numpy()
        # img = transforms.ToPILImage(image) 

        
        trt_output = trt_output.permute(0, 2, 3, 1)
        output = trt_output.cpu().numpy()
        output = output[0]
        output=(output - np.min(output)) / (np.max(output) - np.min(output)) * 255
  
        output = cv2.cvtColor(output, cv2.COLOR_RGB2BGR)
        cv2.imwrite('output.png',output)

**Comment by PolarisKyle** [comment]
trt_img.py与trt_testimg.py都是我用tensorrt的尝试，我的实际情况跟你一样，trt_testimg.py运行快，但效果不行。可能是我调用trt模型的方式不对。

**Comment by fanghaiquan1** [comment]
感谢这么及时的回复！感谢您的分享！

**Comment by fanghaiquan1** [comment]
已经解决了
torchvision.utils.save_image(trt_output, "tu1.jpg")#保存tensor格式为图片

**Comment by fanghaiquan1** [comment]
您的trt_testimg.py程序是正确的，加上tensor格式保存图片就更完美了，再次感谢！
